### PR TITLE
Fix part of #3602: Add Label For ProfileEditActivity

### PR DESCRIPTION
**Pull request by shankarpriyank**
## Explanation
Added label for EditProfileActivity
- "Fixes part of #3602:" 

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

**Comment by shankarpriyank** [comment]
PTAL @bkaur-bkj

File: app/src/main/res/values-ar/strings.xml
Comment (bkaur-bkj) [inline app/src/main/res/values-ar/strings.xml]: @shankarpriyank usually we don't add this from our side, but we can take suggestions from @rt4914.

Comment (shankarpriyank) [reply app/src/main/res/values-ar/strings.xml]: Actually in the last pr which I messed up I was asked to add the string in arabic too

Comment (rt4914) [reply app/src/main/res/values-ar/strings.xml]: @BenHenning Is this correct? We do add strings for other languages too, right?

File: app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileEditActivityTest.kt
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileEditActivityTest.kt]: Remove this blank line.

File: app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileEditActivityTest.kt
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileEditActivityTest.kt:145]: Add one blank line above this.

**Review by rt4914 — CHANGES_REQUESTED** [review]
Almost correct. Suggest minor changes.

File: app/src/main/res/values-ar/strings.xml
Comment (BenHenning) [reply app/src/main/res/values-ar/strings.xml]: We should rename strings when we can, but new strings are added through Translatewiki since we have volunteer translators specifically to help with that task.

File: app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileEditActivityTest.kt
Comment (shankarpriyank) [reply app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileEditActivityTest.kt:145]: Done

File: app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileEditActivityTest.kt
Comment (shankarpriyank) [reply app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileEditActivityTest.kt]: Done

File: app/src/main/res/values-ar/strings.xml
Comment (shankarpriyank) [reply app/src/main/res/values-ar/strings.xml]: @rt4914 Should I undo my changes in the xml file for the Arabic and Portugal language?

**Comment by shankarpriyank** [comment]
PTAL @rt4914

File: app/src/main/res/values-ar/strings.xml
Comment (BenHenning) [reply app/src/main/res/values-ar/strings.xml]: Please do @shankarpriyank. They'll be translated after this PR is merged by contributors on Translatewiki.

File: app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileEditActivityTest.kt
Comment (BenHenning) [inline app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileEditActivityTest.kt:161]: Nit: please remove extra newline.

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @shankarpriyank. Just had a few comments--PTAL.

Also, please update the PR title and description to provide context on what part of #3602 is being fixed by this PR (it's currently not clear).

File: app/src/main/res/values-ar/strings.xml
Comment (shankarpriyank) [reply app/src/main/res/values-ar/strings.xml]: Made the changes as requested , PTAL @rt4914 @BenHenning

**Review by rt4914 — APPROVED** [review]
LGTM, thanks. Just apply suggestions mentioned by Ben.

**Comment by rt4914** [comment]
@shankarpriyank Please reply to all the open comments so that reviewer can know if you have applied the solution or not.

File: app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileEditActivityTest.kt
Comment (shankarpriyank) [reply app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileEditActivityTest.kt:161]: Done

**Comment by shankarpriyank** [comment]
Removed the translations as requested

**Comment by shankarpriyank** [comment]
> @shankarpriyank Please reply to all the open comments so that reviewer can know if you have applied the solution or not.

Doing it

**Review by BenHenning — APPROVED** [review]
Thanks @shankarpriyank. This LGTM for codeowners!

**Review by BenHenning — CHANGES_REQUESTED** [review]
Actually, I just noticed some CI checks are passing. You should look at the error logs for these and make sure that they're all passing @shankarpriyank. PTAL.

**Comment by shankarpriyank** [comment]
Okay , I too had a look at the checks which were failing , I cannot understand why is this happening , could you please guide me through it? @rt4914 @BenHenning

**Comment by shankarpriyank** [comment]
PTAL @BenHenning

**Comment by BenHenning** [comment]
> Okay , I too had a look at the checks which were failing , I cannot understand why is this happening , could you please guide me through it? @rt4914 @BenHenning

@shankarpriyank your test is failing--did you run it locally? You'l need to run it and debug it in order to determine what's wrong. Also, make sure you always run your tests locally to make sure they're passing before pushing changes to GitHub (this will help everything go faster since the CI run can take a while to run, and is slow to debug with; plus it's generally good practice to repeatedly verify that your changed tests are passing throughout development).

**Comment by shankarpriyank** [comment]
@BenHenning  Tried running the test locally , I get this error
Had a look at this [page](https://stackoverflow.com/questions/51262364/kotlin-android-after-updating-kotlin-to-1-2-51-build-fails-execution-failed-f) too but no help
image

**Comment by BenHenning** [comment]
> @BenHenning Tried running the test locally , I get this error Had a look at this [page](https://stackoverflow.com/questions/51262364/kotlin-android-after-updating-kotlin-to-1-2-51-build-fails-execution-failed-f) too but no help image

@shankarpriyank this appears unrelated to the failure from the PR--it actually looks like you can't run tests at all locally. Can you post this as a discussion in https://github.com/oppia/oppia-android/discussions and provide context on your environment, and what you've tried to fix this?

**Comment by shankarpriyank** [comment]
@BenHenning I have started a discussion on the provided platform

**Comment by shankarpriyank** [comment]
Hey @BenHenning I am kind of stuck here, can you please help me a find a way to resolve this

**Comment by BenHenning** [comment]
@shankarpriyank apologies for not following up earlier--are you unblocked now (per the discussion thread)?

**Comment by shankarpriyank** [comment]
@BenHenning The roboelectric tests were running successfully on my local system but some of them failed in the ci check, why are they failing, I still cant figure it out

**Comment by BenHenning** [comment]
Per the logs it looks like org.oppia.android.app.settings.profile.ProfileEditActivityTest is specifically failing @shankarpriyank. Is ``ProfileEditActivityTest`` failing for you locally when run (using Robolectric, not Espresso)?

**Comment by shankarpriyank** [comment]
@BenHenning Actually the suite of tests I was running at that time on my pc were org.oppia.android.app.profile.ProfileChooserFragmentTest(all of them are passing now too , though that's irrelevant here)
Right now when I run org.oppia.android.app.settings.profile.ProfileEditActivityTest it fails saying no tests found
image
 [Gist of the error](https://gist.github.com/shankarpriyank/658cf4cfe00b5ae9d95f1be3d56f1b04)

**Comment by BenHenning** [comment]
@shankarpriyank that's a really unexpected error. Can you try cleaning, invalidating caches, and trying the test again? Also, can you confirm that you trying to run the JUnit version of the test?

**Comment by shankarpriyank** [comment]
@BenHenning It when I ran the tests again after cleaning and invalidating caches, it first threw a java heap out of memory error, and then I changed my settings to this 
image
Though I have got 16gb of ram it wont let me increase the heap sizes anymore than this by throwing an error "Could not reserve ----B of heap size"
And finally I get a different error 
image
 ```
"C:\Program Files (x86)\Java\jdk1.8.0_202\bin\java.exe" -ea -Didea.test.cyclic.buffer.size=1048576 "-javaagent:C:\Program Files\Android\Android Studio\lib\idea_rt.jar=53593:C:\Program Files\Android\Android Studio\bin" -Dfile.encoding=UTF-8 -classpath C:\Users\shank\AppData\Local\Temp\classpath1446411040.jar com.intellij.rt.junit.JUnitStarter -ideVersion5 -junit4 org.oppia.android.app.settings.profile.ProfileEditActivityTest

Process finished with exit code 1
Class not found: "org.oppia.android.app.settings.profile.ProfileEditActivityTest" 
```


And yes they are the test are running in the Junit verison with this config
image

**Comment by BenHenning** [comment]
@shankarpriyank can you verify whether you can run the test via the command line? E.g.:

```sh
./gradlew :app:testDebugUnitTest --tests org.oppia.android.app.settings.profile.ProfileEditActivityTest
```

(This needs to be run from a terminal navigated to the root directory of your Oppia Android repository clone).

**Comment by shankarpriyank** [comment]
@BenHenning Its  not working 
```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':model:compileJava'.
> java.lang.IllegalAccessError: class org.gradle.internal.compiler.java.ClassNameCollector (in unnamed module @0x559f2170) cannot access class com.sun.tools.javac.code.Symbol$TypeSymbol (in module jdk.compiler) because module jdk.compiler does not export com.sun.tools.javac.code to unnamed module @0x559f2170
```

**Comment by BenHenning** [comment]
> java.lang.IllegalAccessError: class org.gradle.internal.compiler.java.ClassNameCollector

@shankarpriyank from a quick search, I found this: https://stackoverflow.com/a/70490234. I think you need to change the Java version your terminal is using (probably by setting the Java home Gradle uses per https://stackoverflow.com/a/37959153 or by changing your Java version in your Linux installation) and then try again.

**Comment by shankarpriyank** [comment]
Hi, @BenHenning  I actually use a Windows environment, but anyways I will do the required and report back.
Right now my semester exams are going on so I am a bit consumed there

**Comment by BenHenning** [comment]
> Hi, @BenHenning I actually use a Windows environment, but anyways I will do the required and report back. Right now my semester exams are going on so I am a bit consumed there

No problem. In that case, the first suggestion for explicitly specifying the Java home for a specific Gradle command should work for you on Windows (updating the main Java version is a bit more onerous and depends on several factors, including whether you're using MinGW or native Windows).

**Comment by shankarpriyank** [comment]
@BenHenning I already have the project configured for java version 1.8, but still I tried using thing command 
gradlew clean build -Dorg.gradle.java.home="C:\Program Files (x86)\Java\jdk1.8.0_202" and it crashes with this message now!
image

Also just for reference
image

**Comment by BenHenning** [comment]
@shankarpriyank what (full) command are you trying to run? I find it a bit odd that the Android linter is running, so I'm wondering whether you're running the correct command.

**Comment by BenHenning** [comment]
Hi. Since I was requested as a reviewer on this PR: Please note that I will be away & unavailable for reviews starting today until 23 May 2022. Please try to make any progress that you can which doesn't require me, and I'll catch up on my reviews when I return. Thanks for your flexibility--I really appreciate it.

**Pull request closed without merging**
